Tools/Algorithms/stencil.py

```diff
@@ -342,68 +354,62 @@
     name : str, optional (default = '')
         Common label for figure names.
     """
+
     # Galilean velocity (default = 0.)
     v_gal = 0.
     if galilean:
         v_gal = -np.sqrt(1.-1./gamma**2)*c
 
     # Display some output
     print('\nCell size:')
-    print('- dx = {:g}'.format(dx))
-    print('- dy = {:g}'.format(dy))
-    print('- dz = {:g}'.format(dz))
-    print('- dz/dx = {:g}'.format(dz/dx))
-    print('- dz/dy = {:g}'.format(dz/dy))
+    print(f'- dx = {dx_boosted}')
+    if dims > 1:
+        print(f'- dx[1:]/dx[0] = {dx_boosted[1:]/dx_boosted[0]}')
     print('\nTime step:')
-    print('- dt = {:g}'.format(dt))
-    print('- c*dt/dx = {:g}'.format(c*dt/dx))
-    print('- c*dt/dy = {:g}'.format(c*dt/dy))
-    print('- c*dt/dz = {:g}'.format(c*dt/dz))
+    print(f'- dt = {dt}')
+    print(f'- c*dt/dx = {c*dt/dx_boosted}')
     print('\nSpectral order:')
-    print('- nox = {:d}'.format(nox))
-    print('- noy = {:d}'.format(noy))
-    print('- noz = {:d}'.format(noz))
+    print(f'- order = {psatd_order}')
     print('\nLorentz boost, Galilean velocity:')
-    print('- gamma = {:g}'.format(gamma))
-    print('- v_gal = {:g}'.format(v_gal))
+    print(f'- gamma = {gamma}')
+    print(f'- v_gal = {v_gal}')
 
-    stencils = compute_all(dx, dy, dz, dt, nox, noy, noz, v_gal)
+    stencils = compute_all(dx_boosted, dt, psatd_order, v_gal)
 
     # Maximum number of cells
-    ncx = 65
-    ncy = 65
-    ncz = 65
-
-    # Array of cell numbers
-    cx = np.arange(ncx)
-    cy = np.arange(ncy)
-    cz = np.arange(ncz)
+    nc = []
+    for i in range(dims):
+        nc.append(65)
 
     # Arrays of stencils
-    stencils['x']['nodal'] = stencils['x']['nodal'][:ncx]
-    stencils['x']['stagg'] = stencils['x']['stagg'][:ncx]
-    stencils['y']['nodal'] = stencils['y']['nodal'][:ncy]
-    stencils['y']['stagg'] = stencils['y']['stagg'][:ncy]
-    stencils['z']['nodal'] = stencils['z']['nodal'][:ncz]
-    stencils['z']['stagg'] = stencils['z']['stagg'][:ncz]
-
-    # Compute minimum number of guard cells for given error threshold
-    # (number of guard cells such that the stencil measure
-    # is not larger than the error threshold)
+    for i, s in enumerate(stencils):
+        s['nodal'] = s['nodal'][:nc[i]]
+        s['stagg'] = s['stagg'][:nc[i]]
+
+    # Axis labels
+    if dims == 1:
+        label = ['x']
+    elif dims == 2:
+        label = ['x', 'z']
+    elif dims == 3:
+        label = ['x', 'y', 'z']
 
     # Plot stencils
-    plot_stencil(cx, stencils['x']['nodal'], stencils['x']['stagg'], 'x', path, name)
-    plot_stencil(cy, stencils['y']['nodal'], stencils['y']['stagg'], 'y', path, name)
-    plot_stencil(cz, stencils['z']['nodal'], stencils['z']['stagg'], 'z', path, name)
+    for i, s in enumerate(stencils):
+        plot_stencil(np.arange(nc[i]), s['nodal'], s['stagg'], label[i], path, name)
```

Comment:
This failure is because the `if` block setting label could leave label undefined if dims is not 1, 2, or 3. Would this be too hard to understand?
`label = ['x', 'y', 'z'][::4-dims]`
Or
```
label = ['x']
if dims == 3: label.append('y')
if dims > 1: label.append('z')
```

Reply:
I went for your second solution, I think it's more readable than the first one, even though slightly more verbose.